app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingService.java

```diff
@@ -262,7 +269,17 @@ public int onStartCommand(Intent intent, int flags, int startId) {
      * function is called to notify this class to run/terminate the syncthing binary.
      * {@link #onServiceStateChange} is called while applying the decision change.
      */
-    private void onUpdatedShouldRunDecision(boolean newShouldRunDecision) {
+    private void onUpdatedShouldRunDecision(RunConditionCheckResult result) {
+        boolean newShouldRunDecision = result.isShouldRun();
+        boolean reasonsChanged;
+        synchronized (mCurrentCheckResult) {
+            reasonsChanged = !mCurrentCheckResult.get().equals(result);
+            mCurrentCheckResult.set(result);
```

Comment:
```
changed = !mCurrentCheckResult.getAndSet(result).equals(result)
```

Reply:
Yes, but in this case we assign the result even if not changed. On the other hand since it is immutable, it doesn't hurt.